src/CoreArbiterServer.cc

```diff
@@ -753,7 +753,11 @@ CoreArbiterServer::timeoutThreadPreemption(int timerFd)
 {
     if (!testingSkipSocketCommunication) {
         uint64_t time;
-        read(timerFd, &time, sizeof(uint64_t));
+        ssize_t ret = read(timerFd, &time, sizeof(uint64_t));
+        if (ret == -1) {
+            LOG(ERROR, "Error reading time: %s", strerror(errno));
```

Comment:
This is actually an error reading "number of expirations of the timer". We should probably update the log message.